x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexServiceTests.java

```diff
@@ -1130,18 +1149,13 @@ public void onFailure(Exception e) {
                 latch.countDown();
             }
         });
-
+        assertTrue("Timeout waiting for delete request", latch.await(TIMEOUT_SECONDS, TimeUnit.SECONDS));
         if (exception.get() != null) {
             throw exception.get();
         }
 
-        boolean requestTimedOut = latch.await(TIMEOUT_SECONDS, TimeUnit.SECONDS);
```

Comment:
I had to move this statement up, as the exception was not re-thrown because the `latch.await` was below it